### PR TITLE
fix(files): Fix type issues of mixing string and number

### PR DESCRIPTION
**Pull request by susnux**
## Summary

Fix some type issues in the files list where numbers were mixed with strings for fileids and thus the strict compare will fail (`===`).

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files/src/components/FileEntry.vue [hunk 1]
```diff
@@ -194,8 +194,8 @@ export default Vue.extend({
 	},
 
 	computed: {
-		currentView(): View {
```
Comment (artonge) [inline apps/files/src/components/FileEntry.vue:197]: Why remove the typing?

File: apps/files/src/components/FileEntry/FileEntryCheckbox.vue [hunk 2]
```diff
@@ -103,8 +110,9 @@ export default Vue.extend({
 
 				const lastSelection = this.selectionStore.lastSelection
 				const filesToSelect = this.nodes
-					.map(file => file.fileid?.toString?.())
+					.map(file => file.fileid)
 					.slice(start, end + 1)
+					.filter(isNumber)
```
Comment (artonge) [inline apps/files/src/components/FileEntry/FileEntryCheckbox.vue:115]: Is it in case the file ID is a string? If so, then this is just hiding the issue, no?

File: apps/files/src/components/FilesListTableHeaderActions.vue [hunk 3]
```diff
@@ -115,13 +119,17 @@ export default Vue.extend({
 		},
 
 		nodes() {
+			// eslint-disable-next-line jsdoc/require-jsdoc
```
Comment (artonge) [inline apps/files/src/components/FilesListTableHeaderActions.vue:122]: Is there a way to disable requiring jsdoc for ts files?

File: apps/files/src/views/FilesList.vue [hunk 4]
```diff
@@ -486,7 +492,9 @@ export default defineComponent({
 			if (window?.OCA?.Files?.Sidebar?.setActiveTab) {
 				window.OCA.Files.Sidebar.setActiveTab('sharing')
 			}
-			sidebarAction.exec(this.currentFolder, this.currentView, this.currentFolder.path)
+			if (this.currentFolder !== undefined) {
```
Comment (skjnldsv) [inline apps/files/src/views/FilesList.vue:495]: Seems like there is a lot of unrelated changes?
How would `this.currentFolder` be undefined?

The FilesList gets rendered only if we have files, which we have only if `this.currentFolder` is defined, no? :thinking:

File: apps/files/src/components/FileEntry.vue [hunk 5]
```diff
@@ -431,14 +435,16 @@ export default Vue.extend({
 
 			// Dragging set of files, if we're dragging a file
 			// that is already selected, we use the entire selection
-			if (this.selectedFiles.includes(this.fileid)) {
-				this.draggingStore.set(this.selectedFiles)
-			} else {
-				this.draggingStore.set([this.fileid])
+			if (this.fileid !== undefined) {
```
[the remark this answers is not in the document]
Comment (susnux) [reply apps/files/src/components/FileEntry.vue:438]: In case `this.source` is undefined / has no fileid, see definition:

```ts
fileid() {
	return this.source?.fileid // this can also be undefined
}
```

File: apps/files/src/components/FileEntry/FileEntryCheckbox.vue (hunk 2 above)
Comment (susnux) [reply apps/files/src/components/FileEntry/FileEntryCheckbox.vue:115]: This is for the case fileid is undefined as the typing of `file.fileid` is `number|undefined`

File: apps/files/src/components/FileEntry.vue (hunk 1 above)
Comment (susnux) [reply apps/files/src/components/FileEntry.vue:197]: Because it was wrong, the correct type would be `View | null` as this is the type of `Navigation.active`

File: apps/files/src/components/FileEntry.vue (hunk 5 above)
[the remark this answers is not in the document]
Comment (susnux) [reply apps/files/src/components/FileEntry.vue]: Makes sense! Changed it and added a comment about the assertion

**Comment by skjnldsv** [comment]
Too much conflicts, closing :see_no_evil:
